examples/template_llava.jinja

```diff
@@ -0,0 +1,11 @@
+{%- for message in messages -%}
+    {{ message['role'].upper() + ': ' + message['content'] }}
+    {%- if (loop.last and add_generation_prompt) or not loop.last -%}
+        {{- '\n' -}}
```

Comment:
I think there should actually be no `'\n'`

Going by the `vicuna_v1` conversation style used for `llava-v1.5-13b` eg [here](https://github.com/haotian-liu/LLaVA/blob/c121f0432da27facab705978f83c4ada465e46fd/scripts/v1_5/eval/llavabench.sh#L9)

Which has `sep2="</s>"`

https://github.com/haotian-liu/LLaVA/blob/c121f0432da27facab705978f83c4ada465e46fd/llava/conversation.py#L242-L252

The initial prompt will look like this:

```py
In [3]: from llava import conversation

In [4]: conv = conversation.conv_vicuna_v1.copy()

In [5]: conv.append_message(conv.roles[0], "Hi")

In [6]: conv.append_message(conv.roles[1], None)

In [7]: conv.get_prompt()
Out[7]: "A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. USER: Hi ASSISTANT:"
```

And then continues like this with a `</s>` after each assistant response.

`In [9]: conv.messages[-1][-1] = " Hello I am LLaVA".strip() # The model will add with space looks like it gets stripped`

`In [10]: conv.get_prompt()
Out[10]: "A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. USER: Hi ASSISTANT: Hello I am LLaVA</s>"`

`In [11]: conv.append_message(conv.roles[0], "What is the capital of France?")`

`In [12]: conv.append_message(conv.roles[1], None)`

`In [13]: conv.get_prompt()`

`Out[13]: "A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. USER: Hi ASSISTANT: Hello I am LLaVA</s>USER: What is the capital of France? ASSISTANT:"`

Reply:
I modeled the chat template according to their [HF repo](https://huggingface.co/llava-hf/llava-1.5-7b-hf). Their example used a newline right before `ASSISTANT`.

Reply:
Interesting, [llama.cpp](https://github.com/ggerganov/llama.cpp/blob/db10f01310beea8a1ef7798651b9d692fd1149d0/examples/llava/llava-cli.cpp#L172-L174) also has the same (before `ASSISTANT` and before the first `USER` after the system prompt) - I guess it is mostly compatible with the original llava style if the user messages could end with a newline during training - although the jinja template also isn't handling the system prompt (which seems is added with no `SYSTEM:` prefix in llama.cpp and LLaVA repo's `conv_vicuna_v1`)

Will (and should?) `</s>` also get added after the ASSISTANT answer? I guess it will have been output from the model since it's the eos token but not sure if it gets stripped at some point before making it to jinja.